blitz_api/tests/tests_view_Users.py

```diff
@@ -0,0 +1,610 @@
+
+    def test_create_new_user_missing_fields(self):
+        """
+        Ensure we can't create a new user without required fields
+        """
+        data = {}
+
+        response = self.client.post(
+            reverse('user-list'),
+            data,
+            format='json',
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+        content = {
+            'academic_field': ['This field is required.'],
+            'academic_level': ['This field is required.'],
+            'birthdate': ['This field is required.'],
+            'email': ['This field is required.'],
+            'first_name': ['This field is required.'],
+            'gender': ['This field is required.'],
+            'last_name': ['This field is required.'],
+            'password': ['This field is required.'],
+            'university': ['This field is required.']
+        }
+        self.assertEqual(json.loads(response.content), content)
+
+    def test_create_new_user_weak_password(self):
+        """
+        Ensure we can't create a new user with a weak password
+        """
+        data = {
+            'username': 'John',
+            'email': 'John@mailinator.com',
+            'password': '19274682736',
+            'first_name': 'Chuck',
+            'last_name': 'Norris',
+            'university': {
+                "name": "random_university"
+            },
+            'academic_field': {'name': "random_field"},
+            'academic_level': {'name': "random_level"},
+            'gender': "M",
+            'birthdate': "1999-11-11",
+        }
+
+        response = self.client.post(
+            reverse('user-list'),
+            data,
+            format='json',
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+        content = {"password": ['This password is entirely numeric.']}
+        self.assertEqual(json.loads(response.content), content)
+
+    def test_create_new_user_invalid_domain(self):
+        """
+        Ensure we can't create a new user with an invalid domain.
+        An invalid domain can be defined as:
+            - Non-existent
+            - Not matching with selected university
+        """
+        data = {
+            'username': 'John',
+            'email': 'John@invalid.com',
+            'password': '1927nce-736',
+            'first_name': 'Chuck',
+            'last_name': 'Norris',
+            'university': {
+                "name": "random_university"
+            },
+            'academic_field': {'name': "random_field"},
+            'academic_level': {'name': "random_level"},
+            'gender': "M",
+            'birthdate': "1999-11-11",
+        }
+
+        response = self.client.post(
+            reverse('user-list'),
+            data,
+            format='json',
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+        content = {'email': ['Invalid domain name.']}
```

Comment:
Un petit bug a ce niveau la il semblerai lors de l'integration Frontend, le message d'erreur n'est pas contenu dans la key `email` mais directement a la racine du message d'erreur ce qui n'est pas standard.

![screenshot from 2018-05-13 15-10-52](https://user-images.githubusercontent.com/12053720/39970811-01c9d932-56c0-11e8-85d5-9730d0e71b0f.png)
![screenshot from 2018-05-13 15-10-56](https://user-images.githubusercontent.com/12053720/39970812-01ec1e2a-56c0-11e8-8af1-db65dcd4bb3c.png)

Reply:
The tests are passing and proper response is received when doing the request to the API manually with Postman. 

Can you reproduce? How?

Reply:
I will try to reproduce with this same data and i keep you inform if i can reproduce